app/jobs/parse_report_job.rb

```diff
@@ -6,11 +6,22 @@
 class ParseReportJob
   include Sidekiq::Worker
 
+  # rubocop:disable Metrics/MethodLength
   def perform(file, message)
     return if cancelled?
 
-    @file = file
     @msg_value = message
+
+    # TODO: clean this up after the legacy jobs are drained in production
+    @file = if file.is_a?(Numeric)
+              SafeDownloader.download_reports(
+                @msg_value['url'],
+                ssl_only: Settings.report_download_ssl_only
+              )[file]
```

Comment:
Observation: Lack of error reporting if the report on a given index does not exist.

Reply:
Right, this would need some additional error management...